packages/pcd/unknown-pcd/src/UnknownPCDPackage.ts

```diff
@@ -0,0 +1,121 @@
+import { DisplayOptions, PCDPackage, SerializedPCD } from "@pcd/pcd-types";
+import { v4 as uuid } from "uuid";
+import {
+  UnknownPCD,
+  UnknownPCDArgs,
+  UnknownPCDClaim,
+  UnknownPCDInitArgs,
+  UnknownPCDProof,
+  UnknownPCDTypeName
+} from "./UnknownPCD";
+
+let savedInitArgs: UnknownPCDInitArgs = undefined;
+
+/**
+ * Initialize UnknownPCDPackage with configuration
+ */
+export async function init(args: UnknownPCDInitArgs): Promise<void> {
+  savedInitArgs = args;
+}
+
+/**
+ * Creates a new {@link UnknownPCD} by generating an {@link UnknownPCDProof}
+ * and deriving an {@link UnknownPCDClaim} from the given {@link UnknownPCDArgs}.
+ *
+ * @throws if the arguments are invalid
+ */
+export async function prove(args: UnknownPCDArgs): Promise<UnknownPCD> {
+  if (!args.serializedPCD.value) {
+    throw new Error("No serialized PCD value provided");
+  }
+
+  // TODO(artwyman): Think harder about this.  There's no way to reliably
+  // extract the proper ID from the SerializedPCD, so it's not going to be
+  // consistent in the PCDCollection.  We could make a best-effort attempt
+  // using JSON and/or JSONBig deserialization to look for an "id" field.
+  // We could also change SerializedPCD going forward to contain an "id"
+  // field, but it would have to be optional.
+  const id = uuid();
```

Comment:
Oh, I should've used derivePCDID() here too, as it's the outcome of this TODO.  It's still not 100% reliable.  I decided against changing the SerializedPCD format, since it seemed too invasive/risky for this PR.

My worry here is that the UnknownPCD doesn't fully "act like" the PCD it's replacing if it has a different ID.  A use case I have in mind primarily PCD replacement by ID, where a wrapped UnknownPCD can be cleanly replaced by another PCD by the same ID rather than leaving behind a mess.  This could come up in feeds if they use Replace without using DeleteFolder first.

Another relevant case is what happens if an UnknownPCD gets re-parsed correctly and reverts to its original ID.  If in the interim a new PCD is put in place, then you end up with two PCDs of the same ID.  The derived ID avoids this when it works, though I think that case ends up behaving reasonably anyway.  The UnknownPCD retains its spot in the insertion order.  That means that the upsert which occurs during deserialization will let the newer PCD replace the older one.  That's not 100% guaranteed to be the intended behavior, but it'll be right most of the time.

Reply:
>  This could come up in feeds if they use Replace without using DeleteFolder first.

yeah this makes the most sense to me